src/backend/gporca/server/src/unittest/CTestUtils.cpp

```diff
@@ -3827,13 +3827,14 @@ CTestUtils::CreateGenericDatum(CMemoryPool *mp, CMDAccessor *md_accessor,
 		CDXLUtils::DecodeByteArrayFromString(mp, pstrEncodedValue, &ulbaSize);
 
 	CDXLDatumGeneric *dxl_datum = NULL;
-	if (CMDTypeGenericGPDB::IsTimeRelatedType(mdid_type))
+	if (CMDTypeGenericGPDB::IsTimeRelatedTypeMappableToDouble(mdid_type))
 	{
 		dxl_datum = GPOS_NEW(mp) CDXLDatumStatsDoubleMappable(
 			mp, mdid_type, default_type_modifier, false /*is_const_null*/, data,
 			ulbaSize, CDouble(value));
 	}
-	else if (pmdtype->IsTextRelated())
+	else if (pmdtype->IsTextRelated() ||
+			 mdid_type->Equals(&CMDIdGPDB::m_mdid_date))
```

Comment:
This "leak of detail" does not look good to me. It breaks the general approach of this sequence of conditional branches, which uses functions do determine "class of type", instead of determining a type itself.

I propose to introduce a method e.g. `IsTimeRelatedTypeMappableToLint()` which would return `mdid->Equals(&CMDIdGPDB::m_mdid_date)` (for now; but it may be expanded in the future).

Reply:
Although, as this is a helper for a test case, leaving everything "as-is" is also acceptable, I suppose.

Reply:
Thanks, added `IsTimeRelatedTypeMappableToLint()`